src/devices/speechSynthesizer_nwc_yarp/SpeechSynthesizer_nwc_yarp.h

```diff
@@ -0,0 +1,137 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2023 Istituto Italiano di Tecnologia (IIT)
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef YARP_DEV_SPEECHSYNTHESIZER_NWC_YARP_H
+#define YARP_DEV_SPEECHSYNTHESIZER_NWC_YARP_H
+
+#include <cstdio>
+#include <cstdlib>
+
+#include <yarp/os/BufferedPort.h>
+#include <yarp/dev/PolyDriver.h>
+#include <yarp/dev/ISpeechSynthesizer.h>
+#include <yarp/sig/Sound.h>
+#include <yarp/os/Time.h>
+#include <yarp/os/Network.h>
+#include <yarp/os/Thread.h>
+#include <yarp/os/Vocab.h>
+#include <yarp/os/Bottle.h>
+#include <yarp/os/RpcServer.h>
+#include <yarp/dev/WrapperSingle.h>
+#include "ISpeechSynthesizerMsgs.h"
+
+
+/**
+ *  @ingroup dev_impl_nwc_yarp
+ *
+ * \section speechSynthesizer_nwc_yarp
+ *
+ * \brief `speechSynthesizer_nwc_yarp`: A network wrapper client that connects to a speechSynthesizer_nws_yarp to perform speech synthesis.
+ *
+ *  Parameters required by this device are:
+ * | Parameter name | SubParameter | Type    | Units          | Default Value | Required     | Description                                                                                           | Notes |
+ * |:--------------:|:------------:|:-------:|:--------------:|:-------------:|:-----------: |:-----------------------------------------------------------------------------------------------------:|:-----:|
+ * | local          | -            | string  | -              | -             | Yes          | Full port name opened by the device.                                                                  |       |
+ * | server         | -            | string  | -              | -             | Yes          | Full port name of the port remotely opened by the server, to which this client connects to.           |       |
+ * | carrier        | -            | string  | -              | tcp           | No           | The carrier used for the connection with the server.                                                   |       |
+ */
+
+class SpeechSynthesizer_nwc_yarp :
+    public yarp::dev::DeviceDriver,
+    public yarp::dev::ISpeechSynthesizer
+{
+    // Methods
+private:
+    bool closeMain();
+
+public:
+    SpeechSynthesizer_nwc_yarp() = default;
+    SpeechSynthesizer_nwc_yarp(const SpeechSynthesizer_nwc_yarp&) = delete;
+    SpeechSynthesizer_nwc_yarp(SpeechSynthesizer_nwc_yarp&&) = delete;
+    SpeechSynthesizer_nwc_yarp& operator=(const SpeechSynthesizer_nwc_yarp&) = delete;
+    SpeechSynthesizer_nwc_yarp& operator=(SpeechSynthesizer_nwc_yarp&&) = delete;
+    ~SpeechSynthesizer_nwc_yarp() override;
+
+    // yarp::dev::DeviceDriver
+    bool open(yarp::os::Searchable& config) override;
+    bool close() override;
+
+    /**
+     * Sets the language for speech synthesis.
```

Comment:
documentation not required, just a comment 'methods from ISpeechSynthesizer' is enough